### PR TITLE
DEVELOPER-969 Ready for Review - Testing events.rb to integrate into Searchisko

### PR DESCRIPTION
**Pull request by Dantheman720**
This is just for demoing the JSON that events.rb creates.

### CODE REVIEW

File: _ext/events.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            else
+              [key, value]
+          end
+        }.to_h
+
+        searchisko_hash.merge!({:unique_id => searchisko_hash['sys_title'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-#{id}"})
+
+        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=#{searchisko_hash[:unique_id]}"})
+
+        ## Set a default priority if none set. 100 is lowest possible priority value.
```
Comment (paulrobinson) [inline _ext/events.rb]: You don't need priority here. That's just on the connectors page to apply a bespoke ordering.

**Comment by Dantheman720** [comment]
I am reading the https://github.com/searchisko/searchisko/blob/master/documentation/rest-api/content/dcp_content_object.md.  It looks like I need to make a few changes before we talk to Lukas

**Comment by Dantheman720** [comment]
It looks like this in the JSON:

{  
   "start_date":"20/01/2015",
   "end_date":"20/01/2015",
   "sys_title":"DeltaSpike: the CDI toolbox",
   "sys_description":"For this Virtual JBUG session we have Antoine Sabot-Durand, covering Apache DeltaSpike. DeltaSpike consists of a number of portable CDI extensions that provide useful features for Java application developers.",
   "sys_content_provider":"Red Hat",
   "more_details":"http://www.meetup.com/JBoss-User-Group-Worldwide/events/218755922/",
   "on_homepage":"TRUE",
   "sys_content_id":"deltaspike-the-cdi-toolbox-20012015",
   "sys_content_type":"jbossorg_event",
   "sys_url_view":"http://localhost:4242/events/event#!id=deltaspike-the-cdi-toolbox-20012015"
}

File: _ext/events.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
+require 'aweplug/helpers/searchisko'
+require 'parallel'
+
+module JBoss
+  module Developer
+    class Events
+
+      def initialize
```
Comment (LightGuard) [inline _ext/events.rb]: We can drop this if there's no impl

File: _ext/events.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+      end
+
+      def execute site
+        searchisko = Aweplug::Helpers::Searchisko.default site, 0
+
+        Parallel.each(site.events, in_threads: (site.build_threads || 0)) do |(id, data)|
+
+        searchisko_hash = data.collect { |(key, value)|
```
Comment (LightGuard) [inline _ext/events.rb]: If it's a multi statement block, use the do ... end syntax

File: _ext/events.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            when 'title'
+              ['sys_title', value]
+            when 'description'
+              ['sys_description', value]
+            when 'type'
+              ['sys_content_provider', value]
+            else
+              [key, value]
+          end
+        }.to_h
```
Comment (LightGuard) [inline _ext/events.rb]: instead of using strings for the hash keys, use symbols

File: _ext/events.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+
+        # Add identifier and other required fields for Searchisko Content Object
+        searchisko_hash.merge!({:sys_content_id => searchisko_hash['sys_title'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')})
+        searchisko_hash.merge!({:sys_content_type => 'jbossorg_event'})
+        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})
```
Comment (LightGuard) [inline _ext/events.rb]: You could do the merge in one step with a larger hash

Comment (Dantheman720) [inline _ext/events.rb]: Ah, I see what you're saying.  Ex: searchisko_hash.merge!({:sys_content_id => searchisko_hash['sys_title'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, ''), :sys_content_type => 'jbossorg_event', :sys_url_view => "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})

I was tabbing between and was negligent on the merge....sorry about that.

Comment (LightGuard) [inline _ext/events.rb]: Yep, though that does become a bit difficult to read, make sure spacing
helps readability, there are a lot of chained methods and three different
key-value pairs being added here.

On Fri, Jan 30, 2015 at 10:12 AM, Dantheman720 <notifications@github.com>
wrote:

> In _ext/events.rb
> <https://github.com/jboss-developer/www.jboss.org/pull/759#discussion_r23855991>
> :
>
> > +            when 'title'
> > +              ['sys_title', value]
> > +            when 'description'
> > +              ['sys_description', value]
> > +            when 'type'
> > +              ['sys_content_provider', value]
> > +            else
> > +              [key, value]
> > +          end
> > +        }.to_h
> > +
> > +
> > +        # Add identifier and other required fields for Searchisko Content Object
> > +        searchisko_hash.merge!({:sys_content_id => searchisko_hash['sys_title'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')})
> > +        searchisko_hash.merge!({:sys_content_type => 'jbossorg_event'})
> > +        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})
>
> Ah, I see what you're saying. Ex: searchisko_hash.merge!({:sys_content_id
> => searchisko_hash['sys_title'].downcase.strip.gsub(' ',
> '-').gsub(/[^\w-]/, '') + "-" +
> searchisko_hash['start_date'].downcase.strip.gsub(' ', '-').gsub(/[^\w-]/,
> ''), :sys_content_type => 'jbossorg_event', :sys_url_view =>
> "#{site.base_url}/events/event#!id=" + searchisko_hash[:sys_content_id ]})
>
> I was tabbing between and was negligent on the merge....sorry about that.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jboss-developer/www.jboss.org/pull/759/files#r23855991>
> .
>



-- 
Jason Porter
http://en.gravatar.com/lightguardjp

File: _ext/events.rb (hunk 3 above)
Comment (Dantheman720) [inline _ext/events.rb]: I'm confused on this one.  Which block are you referring to?

Comment (LightGuard) [inline _ext/events.rb]: This whole { ... } block, please switch it to use the do ... end syntax. The { ... } syntax is for one liners.

File: _dcp/data/provider/jboss-developer.json [hunk 6]
```diff
@@ -427,7 +427,7 @@
       "description":"JBoss Developer website pages",
       "sys_type":"webpage",
       "input_preprocessors": [
-        { 
+        {
```
Comment (paulrobinson) [inline _dcp/data/provider/jboss-developer.json]: Try not to make white-space changes in un-changed areas of code. It makes it harder to detect what material changes you have made, when reviewing the PR. I think it would probably also increase the chances of merge conflicts.

File: _dcp/data/provider/jboss-developer.json [hunk 7]
```diff
@@ -270,7 +270,7 @@
               }
             ]
           }
-        },     
+        },
```
Comment (paulrobinson) [inline _dcp/data/provider/jboss-developer.json]: See below about ws changes

File: _ext/pipeline.rb [hunk 8]
```diff
@@ -21,12 +21,13 @@
 require 'disqus_more'
 require 'lower_case_paths'
 require 'customer_portal'
+require 'events'
 require 'connectors'
 require 'vault'
 require 'aweplug/helpers/define'
 
 Awestruct::Extensions::Pipeline.new do
-
```
Comment (paulrobinson) [inline _ext/pipeline.rb]: Lots ow WS changes in this file. Take a look at the diff, you'll see how difficult it is to find your actual change.

**Comment by Dantheman720** [comment]
I consolidated the diff.

**Comment by paulrobinson** [comment]
Have you applied all the reviewer's fixes, including those in this PR and the email conversation?

File: _dcp/data/provider/jboss-developer.json [hunk 9]
```diff
@@ -466,6 +466,20 @@
         "name":"data_jbossdeveloper_connector",
         "type":"jbossdeveloper_connector"
       }
+    },
+
+    "jbossdeveloper_events": {
```
Comment (lukas-vlcek) [inline _dcp/data/provider/jboss-developer.json]: Please use singular: `jbossdeveloper_event`

File: _dcp/data/provider/jboss-developer.json [hunk 10]
```diff
@@ -466,6 +466,20 @@
...
+      "description":"Events",
+      "sys_type":"event",
+      "sys_content_content-type" : "text/plain",
+      "persist" : true,
+      "input_preprocessors": [
+      ],
+      "index": {
+        "name":"data_jbossdeveloper_events",
```
Comment (lukas-vlcek) [inline _dcp/data/provider/jboss-developer.json]: We need to use singular here as well: `data_jbossdeveloper_event`
It needs to match with the name of file `_dcp/indexes/data_jbossdeveloper_event.json` (without the extension)

File: _dcp/data/provider/jboss-developer.json [hunk 11]
```diff
@@ -466,6 +466,20 @@
...
+        "type":"jbossdeveloper_events"
```
Comment (lukas-vlcek) [inline _dcp/data/provider/jboss-developer.json]: We need to use singular here as well: `jbossdeveloper_event`
It needs to match with the name of file (without extension): `_dcp/mappings/data_jbossdeveloper_event/jbossdeveloper_event.json`

File: _dcp/mappings/data_jbossdeveloper_event/jbossdeveloper_event.json [hunk 12]
```diff
@@ -0,0 +1,11 @@
+{
+  "jbossdeveloper_event": {
+    "properties": {
+      "sys_title": {
+        "type": "string",
+        "include_in_all": true,
+        "_boost": 1.5
+      }
```
Comment (lukas-vlcek) [inline _dcp/mappings/data_jbossdeveloper_event/jbossdeveloper_event.json]: We should create mapping for both `start_date` and `end_date`. Add the following into `properties`:

    "start_date": { "type": "date" },
    "end_date": { "type": "date" }

File: events.html.slim [hunk 13]
```diff
@@ -47,28 +47,29 @@ description: Discover the events and webinars featuring talks about JBoss produc
...
-              p=e['description']
-              - if "red hat".casecmp(e['type']) == 0
-                p.red-hat-event
-                  i.icon-redhat
-                  | Red Hat Event
-            .event-links
-              a href=e['more_details']
-                | Learn More
-            = partial("share.html.slim", :url => e['more_details'], :title => e['description'][0 .. 139])
+        - if DateTime.parse(e['end_date']).to_time > DateTime.now.prev_day.to_time.localtime("+09:00")
```
Comment (paulrobinson) [inline events.html.slim]: This needs to be in a separate PR so that we can merge it immediately. It's a stop-gap whilst we wait for the work in this PR to be complete.

File: _ext/events.rb [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+          end
+        end.to_h
+
+        id_gen = searchisko_hash[:sys_title].to_s.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '') + "-" + searchisko_hash['start_date'].to_s.downcase.strip.gsub(' ', '-').gsub(/[^\w-]/, '')
+
+        # Add identifier and other required fields for Searchisko Content Object
+        searchisko_hash.merge!({:sys_url_view => "#{site.base_url}/events/event#!id=" + id_gen.to_s})
+
+        unless !site.push_to_searchisko || site.profile =~ /development/
+          searchisko.push_content('jbossdeveloper_events',
```
Comment (paulrobinson) [inline _ext/events.rb]: Should this be 'jbossdeveloper_event' (no plural). Notice what is used elsewhere.

**Comment by paulrobinson** [comment]
retest this please

**Comment by paulrobinson** [comment]
The event content type is now available in DCPBeta. However, it looks like the wrong date format is being used. I suggest you look at other code that's pushing dates and see what format it uses and how we instruct Ruby to output it.
